### PR TITLE
#873: add Dashboard widget

### PR DESCRIPTION
**Pull request by raduiordache**
- added dashboard widget function in widget manager
- wrote test to check
- added enums related to it

Fixes #<issue>

## Checklist:
- [x] My code follows the pykechain style
    - [x] I added type hinting to all functions
    - [x] I added documentation for all public functions
    - [x] I locally used `tox` to check for dists and docs errors
    - [x] My code passes the `pep8` and `flake8` linting checks and no warnings
    - [x] I removed unused imports, using `Code > Optimize Imports` in PyCharm
    - [x] I used [`black`](https://github.com/psf/black) to format the new code
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] I committed fresh test cassettes
    - [x] I have proper test coverage (don't decline the coverage of the test)
- [x] I asked another teammate to review the code
- [x] I update the Changelog.md with the appropriate changes.

### CODE REVIEW

**Review by JelleBoersma — APPROVED** [review]
Very nice! I see no apparent issues.

File: pykechain/models/widgets/widgets_manager.py [hunk 1]
```diff
@@ -1449,6 +1451,157 @@ def add_service_card_widget(
...
+
+        The widget will be saved in KE-chain
+
+        :param title: A custom title for the card widget
+            * False (default): Card name
+            * String value: Custom title
+            * None: No title
+        :param parent_widget: (O) parent of the widget for Multicolumn and Multirow widget.
+        :type parent_widget: Widget or basestring or None
+        :param source_projects: The `Project(s)` to be used as source when displaying the Widget.
```
Comment (jberends) [inline pykechain/models/widgets/widgets_manager.py:1484]: do we want to call this projects? or Scopes.

In the schema there are these 'source' options:

```json
"source": {
  "type": "string",
  "enum": [
    "scopes",
    "selectedScopes",
    "taggedScopes",
    "project",  # inconsistent!
    "subProcess"
  ]
}
```
Maybe ambiguous as well, but predominantly we use scopes everywhere I would say in development and for endusers we call it Projects.

It is better to constrain ourselve to the use of `scope` in pykechain as much as we can.

Also 'defaults on `CURRENT_PROJECT` should be renamed to --> `CURRENT_SCOPE`

**Review by jberends — CHANGES_REQUESTED** [review]
I would consider fixing the suggested changes.

**Comment by jberends** [comment]
for reference, this is the dashboard widget_meta

```
{
            "schema_version": "1.0.0",
            "widget_type": "DASHBOARD",
            "meta_schema": {
                "$schema": "http://json-schema.org/draft-07/schema#",
                "title": "Dashboard Widget Meta Schema",
                "version": "1.0.0",
                "type": "object",
                "additionalProperties": false,
                "properties": {
                    "showTitleValue": {
                        "type": "string",
                        "enum": [
                            "No title",
                            "Custom title",
                            "Default"
                        ]
                    },
                    "customTitle": {
                        "$ref": "#/definitions/nullString"
                    },
                    "customHeight": {
                        "$ref": "#/definitions/positiveInteger"
                    },
                    "collapsed": {
                        "$ref": "#/definitions/booleanNull"
                    },
                    "collapsible": {
                        "$ref": "#/definitions/booleanNull"
                    },
                    "noBackground": {
                        "$ref": "#/definitions/booleanNull"
                    },
                    "noPadding": {
                        "$ref": "#/definitions/booleanNull"
                    },
                    "isDisabled": {
                        "$ref": "#/definitions/booleanNull"
                    },
                    "isMerged": {
                        "$ref": "#/definitions/booleanNull"
                    },
                    "source": {
                        "type": "string",
                        "enum": [
                            "scopes",
                            "selectedScopes",
                            "taggedScopes",
                            "project",
                            "subProcess"
                        ]
                    },
                    "subProcessId": {
                        "$ref": "#/definitions/nullUuidString"
                    },
                    "scopeList": {
                        "type": "array",
                        "items": {
                            "$ref": "#/definitions/nullString"
                        }
                    },
                    "scopeTag": {
                        "type": "array",
                        "items": {
                            "$ref": "#/definitions/nullString"
                        }
                    },
                    "showNumbers": {
                        "type": "array",
                        "items": {
                            "name": {
                                "$ref": "#/definitions/nullString"
                            },
                            "selected": {
                                "$ref": "#/definitions/booleanNull"
                            },
                            "order": {
                                "$ref": "#/definitions/positiveInteger"
                            }
                        }
                    },
                    "showNumbersProjects": {
                        "type": "array",
                        "items": {
                            "name": {
                                "$ref": "#/definitions/nullString"
                            },
                            "selected": {
                                "$ref": "#/definitions/booleanNull"
                            },
                            "order": {
                                "$ref": "#/definitions/positiveInteger"
                            }
                        }
                    },
                    "showOpenVsClosedTasks": {
                        "$ref": "#/definitions/booleanNull"
                    },
                    "showOpenTaskAssignees": {
                        "$ref": "#/definitions/booleanNull"
                    },
                    "showAssignees": {
                        "$ref": "#/definitions/booleanNull"
                    },
                    "showOpenClosedTasksAssignees": {
                        "$ref": "#/definitions/booleanNull"
                    },
                    "showAssigneesTable": {
                        "$ref": "#/definitions/booleanNull"
                    },
                    "activityId": {
                        "$ref": "#/definitions/nullUuidString"
                    }
                },
                "definitions": {
                    "uuidString": {
                        "type": "string",
                        "pattern": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$"
                    },
                    "nullUuidString": {
                        "type": [
                            "string",
                            "null"
                        ],
                        "pattern": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$"
                    },
                    "nullString": {
                        "type": [
                            "string",
                            "null"
                        ]
                    },
                    "positiveInteger": {
                        "type": [
                            "integer",
                            "null"
                        ],
                        "minimum": 0
                    },
                    "booleanNull": {
                        "type": [
                            "boolean",
                            "null"
                        ]
                    }
                }
            }
        },```

**Review by jberends — APPROVED** [review]

**Review by jberends — APPROVED** [review]

**Pull request merged**